### DISCUSSION TITLE
Combine code generator with code fix

### DISCUSSION THREAD

**Discussion by Superheftig** [dotnet/roslyn#53899] [Q&A]
I wrote a code generator that also creates diagnostics, if the analyzed code is somehow "wrong". I also wrote a code fix for the diagnostic id, but it seems not to be picked up (So i don´t get the suggestion from visual studio). If i write an additional DiagnosticAnalyzer and remove it from the code generator for the error case the code fix works.

Is it not possible to combine a code generator with a code fix?

**Comment by huoyaoyuan** [comment]
Considering execution order. It should be achievable, since generated code needs to be contained to obtain a complete semantic model.

**Reply by Superheftig** [reply]
It works now! Seems that I did not configured something correctly :)
